Bottom: 19649da1d8a5e31f9ff46d40fbe2faaa6e67cfb3
Top:    23d5b707519e721112ee61898d8bfa06065815f7
Author: Jason Wang <jasowang@redhat.com>
Date:   2017-09-02 13:18:48 +0800

Refresh of vhost-prefetch-vring

---

diff --git a/drivers/vhost/vhost.c b/drivers/vhost/vhost.c
index 227fcad..5aa93b5 100644
--- a/drivers/vhost/vhost.c
+++ b/drivers/vhost/vhost.c
@@ -2525,11 +2525,12 @@ int vhost_prefetch_desc_indices(struct vhost_virtqueue *vq,
 		total -= copied;
 	}
 
-	if (cont) {
+
+	if (*cont) {
 		__virtio16 ring_head = vhost16_to_cpu(vq, heads[0].id);
 		ret2 = vhost_copy_from_user(vq, descs,
 					    vq->desc + ring_head,
-					    sizeof descs[0]);
+					    ret * sizeof descs[0]);
 		if (unlikely(ret2)) {
 			vq_err(vq, "Failed to get descriptor\n");
 			return -EFAULT;
